bluesky/run_engine.py

```diff
@@ -91,6 +91,8 @@ def __set__(self, obj, value):
      Parameter('subs', Parameter.POSITIONAL_ONLY, default=None),
      Parameter('metadata_kw', Parameter.VAR_KEYWORD)])
 
+def default_scan_id_source(md):
+    return md.get('scan_id', 0) + 1
```

Comment:
The jsonschema validation will check that because that schema requires that, if `scan_id` provided it must be an integer: https://github.com/NSLS-II/event-model/blob/master/event_model/schemas/run_start.json#L19

I think I'd prefer to just check it there, in one place. One could argue this goes against our rule of "fail early", but the jsonschema valdation happens in the same function -- not _that_ much later -- so I think we can just rely on that. Seems reasonable?

Reply:
Thanks for the details about the validator, @danielballan. Yes, I agree on that approach.